corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -723,30 +714,27 @@ def generate_identifier(requests, identifier_type):
     page, and the resulting authenticated session used for sending
     requests.
     """
-    with Session() as session:
-        authenticate_session(requests, session)
+    identifier = None
+    source_id = None
+    with session_context(requests):
```

Comment:
Could be `with requests.session as session` ?

Reply:
Yeah I think that's much better. I've never written a context manager as a method, but I'm sure it's just the same as writing it as a function. I'll play with that and see what works.